src/main/java/net/minecraftforge/client/model/pipeline/RemappingVertexPipeline.java

```diff
@@ -79,26 +103,39 @@ public VertexConsumer uv2(int u, int v)
         return this;
     }
 
+    @Override
+    public VertexConsumer misc(VertexFormatElement element, int... values)
+    {
+        Integer id = miscElementIds.get(element);
+        if (id != null)
+            misc[id] = values;
+        return this;
+    }
+
     @Override
     public void endVertex()
     {
         for (var element : targetFormat.getElements())
         {
-            switch (element.getUsage())
-            {
-                case POSITION -> parent.vertex(position.x, position.y, position.z);
-                case NORMAL -> parent.normal(normal.x(), normal.y(), normal.z());
-                case COLOR -> parent.color(color[0], color[1], color[2], color[3]);
-                case UV ->
-                {
-                    switch (element.getIndex())
-                    {
-                        case 0 -> parent.uv(uv0[0], uv0[1]);
-                        case 1 -> parent.overlayCoords(uv1[0], uv1[1]);
-                        case 2 -> parent.uv2(uv2[0], uv2[1]);
-                    }
-                }
-            }
+            // Ignore padding
+            if (element.getUsage() == VertexFormatElement.Usage.PADDING)
+                continue;
+
+            // Try to match and output any of the supported elements, and if that fails, treat as misc
+            if (element.equals(DefaultVertexFormat.ELEMENT_POSITION))
```

Comment:
Wouldn't a reference comparison be more appropriate here? The elements are constants anyway.

Reply:
The vanilla ones are, but there's a chance someone might make an equivalent element that isn't the same object, and this should match those by usage/index/size as well.